src/components/Navigator.js

```diff
@@ -394,6 +398,11 @@ class Navigator extends React.Component {
   };
 
   componentDidMount = () => {
+    this.setState({
+      selectedFileId : this.props.currentOpenFileId.sourceId,
+      selectedViewId: this.props.currentOpenFileId.viewId,
+      selectedFileOpen: this.props.currentOpenFileId.viewId != 0
```

Comment:
Maybe I misunderstand how this works, but why do we check the `viewId`. Wouldn't the file being open be dependent on the `sourceId` being valid?